docs/core-concepts/app-types.md

```diff
@@ -0,0 +1,125 @@
+property on that dependency means that at publish time, the tooling will skip publishing the assemblies for that dependency 
+to the published output. You don't need these since they will be installed with .NET Core on the targeted machine. 
+
+### Portable application with native dependencies
+A sub-group of the above, this type is a portable application that has native dependencies specified 
+somewhere in its dependency chain. This application is as portable as all of its native dependencies 
+are portable. You will be able to run the application on any platform that your native dependencies can 
+run on. Prime example of this is Kestrel, the ASP.NET Core cross-platform web server. It is built on top of 
+[libuv](https://github.com/libuv/libuv) which is its native dependency. 
+
+When you publish a portable application that has a native dependency, the published output will contain 
+all the same things as the portable application described in the previous section. For native dependencies
+the published output will contain a folder for each [RID](#what-are-rids) that the native dependency supports 
+(and that exists in its NuGet package). 
+
+As said, the portable application with a native dependency is the same as a portable application, just with a 
+dependency that has native components. The sample `project.json` fragment below is showing that:
+
+```json
+"dependencies": {
+    "Microsoft.NETCore.App": {  
+        "version": "1.0.0-*",
+        "type": "platform"
+    }
+    "Microsoft.AspNetCore.Server.Kestrel": "1.0.0-*"
+},
+"frameworks": {
+    "netcoreapp1.0": {}
+}
+```
+
+## Self-contained application
+Unlike the portable application, a *self-contained application* does not rely on any shared component to 
+be present on the machine where you want to deploy the application. As its name implies, it means that 
+the entire dependency closure, **including the runtime** is packaged with the application. This makes 
+it larger, but also makes it capable of running on any .NET Core supported platforms with the correct 
+native dependencies, regardless if it has .NET Core installed or not. This makes it that much 
+easier to deploy to the target machine, since you only deploy your application. 
+
+Since the application carries the runtime with itself, you need to make an explicit choice which platforms your application 
+needs to run on. For instance, if you publish a self-contained application for Windows 10, that same application will 
+not work on OS X or Linux and vice versa. Of course, you can add or remove platforms during development at any given time. 
+
+There are several steps to get to a self-contained application. The first is to remove any `"type": "platform"` properties 
+off of any dependencies you have. Second is to leave the dependency on `Microsoft.NETCore.App` as it will pull in
+all of the rest of things that are needed. 
+
```

Comment:
This sentence ("second is to leave..") has bugged me since I first read this, and it still bugs me. Instructing the user to take an action which is to in fact take no action doesn't make sense. If you feel it is vital to communicate that apps need Microsoft.NETCore.App, perhaps a separate section explaining what the meta package is, what it contains, and what it means to have a type attribute on it would be a better way to go.